tools/serverpod_cli/lib/src/analyzer/models/validation/restrictions/scope.dart

```diff
@@ -25,13 +25,18 @@ class ScopeValueRestriction
     var field = document.findField(parentNodeName);
     if (field == null) return [];
 
-    if (value != ModelFieldScopeDefinition.all && !field.type.nullable) {
-      return [
-        SourceSpanSeverityException(
-          'The field "$parentNodeName" must be nullable when the "${Keyword.scope}" property is set to "${value.name}".',
-          span,
-        )
-      ];
+    var nullableErrorMessage = SourceSpanSeverityException(
+      'The field "$parentNodeName" must be nullable when the "${Keyword.scope}" property is set to "${value.name}".',
+      span,
+    );
+    if (document.serverOnly) {
+      if (value == ModelFieldScopeDefinition.none && !field.type.nullable) {
+        return [nullableErrorMessage];
+      }
+    } else {
+      if (value != ModelFieldScopeDefinition.all && !field.type.nullable) {
```

Comment:
Suggestion: Combine these if statements with && and `else if` as that makes this more clear and we eliminate the implicit else in the inner if statement (meaning less branching).

Reply:
Fixed, we might gain some performance here but IMO this makes the code a bit harder to follow.